### PR TITLE
Test to make sure number of files accepted by the new parser only grows

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

A new test to verify that some `Standard.Base` library files can be parsed by the new parser is added. Use it as:
```bash
$ sbt bootstrap
$ sbt buildEngineDistribution
$ sbt "runtime/testOnly *ParseStdLibTest"
```
By having this test instantly available we can block integration of PRs that would decrease the number of parseable files or introduce new files that aren't paseable.

### Important Notes

The test contains a _black list_ of classes that are currently known to fail. Over the time the _black list_ shall get smaller and smaller.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+                return false;
+            case "Json.enso":
+                return false;
+            case "Bound.enso":
+                return false;
+            case "Rank_Method.enso":
+                return false;
+            case "Regression.enso":
+                return false;
+        }
```
Comment (radeusgd) [inline engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java]: This looks a bit scary.

Can't we do `private static final List<String> blacklist = List.of("Regression.enso", ...)` and then just `blacklist.contains`?

**Review by kazcw — APPROVED** [review]
Thank you!

Could it print when a blacklisted file is found to succeed, so that it's easier to update it?

File: engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        TestSuite s = new TestSuite();
+        collectDistribution(s, "Base");
+        return s;
+    }
+
+    private static Path locateDistribution(final String name) throws URISyntaxException {
+        var where = new File(ParseStdLibTest.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+        var dir = where;
+        for (;;) {
+            dir = new File(new File(new File(new File(new File(new File(where, "distribution"), "lib"), "Standard"), name), "0.0.0-dev"), "src");
```
Comment (hubertp) [inline engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java]: It's 2022 and one can still not write `dir / a / b / c` in Java (but can in Scala!) :)

File: engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (!file.getFileName().toString().endsWith(".enso")) {
+                    return FileVisitResult.CONTINUE;
+                }
+                final String name = file.getFileName().toString();
+                if (isKnownToWork(name)) {
```
Comment (hubertp) [inline engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java]: Potentially write an opposite one that one can quickly run to discover what has improved

**Review by hubertp — APPROVED** [review]
Minus the comments from other reviewers

File: engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java (hunk 2 above)
Comment (JaroslavTulach) [reply engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java]: One can do [59f3db7](https://github.com/enso-org/enso/pull/3814/commits/59f3db7a01f19718256bd4d4f0fe972d2dd4e0c5) when one isn't lazy.

File: engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java (hunk 3 above)
Comment (JaroslavTulach) [reply engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java]: Running all the tests as of [cdd1e6c](https://github.com/enso-org/enso/pull/3814/commits/cdd1e6c95cecc2c9b4f1aca5ed2c0574c144b001) and checking whether they succeed/fail as expected.

File: engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime/src/test/java/org/enso/compiler/ParseStdLibTest.java]: Done in [5fde226](https://github.com/enso-org/enso/pull/3814/commits/5fde22602199b6966e88781a043f608e531556d7)

**Comment by JaroslavTulach** [comment]
> Could it print when a blacklisted file is found to succeed, so that it's easier to update it?

Since [5fde226](https://github.com/enso-org/enso/pull/3814/commits/5fde22602199b6966e88781a043f608e531556d7) there are going to be test failures when a blacklisted file is found to succeed.

**Comment by JaroslavTulach** [comment]
[Windows run](https://github.com/enso-org/enso/actions/runs/3294929899/jobs/5432956891) fails. Why!? Trying to build Enso on Windows to reproduce the problem.... given up on trying to get Rust running on Windows. Rather making the test more verbose: [ba7fba7](https://github.com/enso-org/enso/pull/3814/commits/ba7fba7375dd31b31d9b267ab2898c893927c6ca) - possibly we'll be able to deduce what is wrong from the enhanced output.

**Pull request merged**